client/coral-admin/src/actions/settings.js

```diff
@@ -21,10 +23,16 @@ export const fetchSettings = () => dispatch => {
     });
 };
 
+// for updating top-level settings
 export const updateSettings = settings => {
   return {type: SETTINGS_UPDATED, settings};
 };
 
+// this is a nested property, so it needs a special action.
+export const updateWordlist = (listName, list) => {
```

Comment:
This makes sense, esp with the `metadata` stuff in the future. I'd just say let's keep it somewhat sane and not have stuff like:

```
settings = {
  top_level_prop: {
    sub_prop: [{ // array item that we have to update by index
      please_dont_do_this: String,
      for_the_love_of_all_that_is_holy: Number
   }],
  }
}
```